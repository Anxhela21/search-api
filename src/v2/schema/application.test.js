/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2020. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('Application Resolver', () => {
  test('Correctly Resolves Application Query', (done) => {
    // TODO - Include policy in test query
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
          {
            applications {
              uid
              name
              namespace
              clusters {
                uid
                name
                namespace
              }
              managedSubscriptions {
                uid
                name
                namespace
              }
              policies {
                uid
                name
                namespace
              }
              created
            }
          }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
});
