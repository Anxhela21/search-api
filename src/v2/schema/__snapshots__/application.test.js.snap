// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Resolver Correctly Resolves Application Query 1`] = `
Object {
  "data": Object {
    "applications": Array [
      Object {
        "_uid": "local-cluster/1d4bb419-3666-11ea-9e7f-00000a100f99",
        "created": "2020-01-14T00:37:59Z",
        "dashboard": null,
        "hubSubs": "Failed=1;Propagated=2",
        "name": "samplebook-gbapp",
        "namespace": "sample",
        "pods": "Running=6",
        "remoteCls": 1,
        "remoteSubs": "Failed=1;Propagated=2",
      },
    ],
  },
}
`;
